cache.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/bitrise-init/scanners/android"
+	"github.com/bitrise-io/bitrise-init/utility"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	androidCache "github.com/bitrise-steplib/bitrise-step-android-unit-test/cache"
+)
+
+type depsFunc func(dir string) ([]string, []string, error)
+
+var depsFuncs = []depsFunc{
+	cocoapodsDeps,
+	carthageDeps,
+	androidDeps,
+}
+
+func cocoapodsDeps(dir string) ([]string, []string, error) {
```

Comment:
cocoapodsDeps and carthageDeps does roughly the same, code duplication could be removed.